fivox/eventSource.cpp

```diff
@@ -125,26 +215,23 @@ Events EventSource::findEvents( const AABBf& area LB_UNUSED ) const
         _impl->rtree.query( bgi::intersects( query ), std::back_inserter( hits ));
         maxHits = std::max( size_t(maxHits), hits.size( ));
 
-        Events events;
-        events.reserve( hits.size( ));
+        eventValues.reserve( hits.size( ));
         for( const Value& value : hits )
-        {
-            const Event& val = _impl->events[ value.second ];
-            if( val.value != VALUE_UNSET )
-                events.push_back( val );
-        }
-        return events;
+            eventValues.push_back( getValues()[value.second] );
     }
+    else
 #endif
-
-    static bool first = true;
-    if( first )
+    // return empty
     {
-        LBWARN << "slow path: rtree acceleration not available for findEvents"
-               << std::endl;
-        first = false;
+        static bool first = true;
+        if( first )
+        {
+            LBWARN << "RTree not available for findEvents. "
+                   << "No events will be returned" << std::endl;
+            first = false;
+        }
```

Comment:
We can, but it's only used in the frequency/density functors anyway. For the rest is not worth it.
Also, even if we do that, this 'else' branch you quote needs to remain, in case the rtree hasn't been build before calling findEvents().